packages/block-library/src/table/edit.js

```diff
@@ -500,15 +506,36 @@ export class TableEdit extends Component {
 					/>
 				</InspectorControls>
 				<figure className={ className }>
-					<table className={ tableClasses }>
+					<table className={ tableClasses } aria-labelledby={ captionId }>
 						<Section type="head" rows={ head } />
 						<Section type="body" rows={ body } />
 						<Section type="foot" rows={ foot } />
 					</table>
+					{ /*
+					   * Use a figcaption, which is visibly hidden and a separate RichText for updating
+					   * the caption. Screenreaders seem to have issues announcing the caption of a
+					   * when rendered using a RichText due to the extra wrapping divs added by the RichText.
+					   * Using two elements is a workaround for the problem.
+					   */ }
+					<figcaption id={ captionId } className="screen-reader-text">{ caption }</figcaption>
+					<RichText
+						className={ classnames( 'wp-block-table__caption-content', {
+							'is-visible': isSelected || caption,
+						} ) }
+						tagName="div"
+						placeholder={ __( 'Write caption…' ) }
+						value={ caption }
+						onChange={ ( value ) => setAttributes( { caption: value, captionId } ) }
+						// Deselect the selected table cell when the caption is focused.
+						unstableOnFocus={ () => this.setState( { selectedCell: null } ) }
+					/>
```

Comment:
I really don't like this hack with the separate `figcaption` and `RichText` elements. I'm not sure if this was a known issue in the image block and accepted that it wouldn't work.

Reply:
Why not wrap the `<RichText>` instance in a `<figure>` element, and use a `div` as the `tagName`? One day we'll no longer have all the wrapping tags, but until that day we could do it like that?

Reply:
> One day we'll no longer have all the wrapping tags

Would love you to have a look at #17607, which will allow the table block to omit the wrapper div.